streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -662,6 +664,16 @@ object StreamingContext extends Logging {
     }
   }
 
+  /**
+   * Get the context that is currently active, that is, started but not stopped.
+   */
+  @Experimental
+  def getActive(): Option[StreamingContext] = {
+    ACTIVATION_LOCK.synchronized {
+      Option(activeContext.get())
```

Comment:
I think we should use `ACTIVATION_LOCK` for `StreamingContext.start` and `stop`. At least, `setActiveContext` and `StreamingContext.state` should be protected by `ACTIVATION_LOCK`.